modules/manage/partials/authentication.adoc

```diff
@@ -0,0 +1,1661 @@
+
+===== OIDC credentials flow and access token validation
+
+Before configuring OIDC, you should understand the credentials flow, and in particular,
+the validation claims included in the access token, as you will need to provide them in
+the OIDC configuration.
+
+Redpanda's implementation of OIDC adheres to the Client Credentials Flow defined in
+https://datatracker.ietf.org/doc/html/rfc6749#section-4.4[OAuth 2.0 RFC 6749, section 4.4^]
+in which a client obtains an access token from the authorization server, and provides this
+access token to Redpanda, either using SASL/OAUTHBEARER for the Kafka API, or an HTTP
+Authorization (Bearer) header.
+
+The access token is a _bearer token_. A bearer token is used for authentication and
+authorization in web applications and APIs, and holds user credentials, usually in the form
+of random strings of characters. Bearer tokens are generated based on protocols and specifications
+such as JWT (JSON Web Token), which has a header, payload, and signature. The signature must be
+verified according to the JWK. Claims inside the token and the token signature must both be
+validated. After validation, a configurable claim from the token payload is extracted as
+the principal and attached to the connection, as with any other authentication method.
+
+Following is an example JWT header:
+
+[,json]
+----
+{
+  "alg": "RS256",
+  "typ": "JWT",
+  "kid": "tMQzailSAdaW4nojXxES9"
+}
+----
+
+Following is an example JWT payload:
+
+[,json]
+----
+{
+  "iss": "https://dev-ltxchcls4igzho78.us.auth0.com/",
+  "sub": "3JJeI4tmMC6v8mCVCSDnAGVf2vrnJ0BT@clients",
+  "aud": "localhost",
+  "iat": 1694430088,
+  "exp": 1694516488,
+  "azp": "3JJeI4tmMC6v8mCVCSDnAGVf2vrnJ0BT",
+  "scope": "email2",
+  "gty": "client-credentials"
+}
+----
+
+Following are additional validation claims (JWT properties) that are included in the access token:
+
+* `alg`: The signature algorithm. The extension point in the JWT header is the signature algorithm used to sign the token, and cannot contain the value `none`.
+* `aud`:  Audience. Must match the configuration specified in `oidc_token_audience`. Cannot contain the value `none`.
+* `kid`: Key identifier. Must match _any_ of the public JWK listed in the `jwks_uri` endpoint.
+* `exp`: Expiration. The timestamp listed is greater than current time. Must validate within acceptable bounds of the value specified in `oidc_clock_skew_tolerance`. A clock skew tolerance period may be configured by an Admin to account for clock drift between Redpanda and the OIDC Identity Provider (IdP).
+* `iss`: Issuer. Must exactly match the `issuer` property of the JSON returned from the URL specified in `oidc_discovery_url`.
+* `scope`: Scope. Must include the value `openid`.
+* `sub`: Subject. Must match exactly the Principal ID found through the principal mapping.
```

Comment:
`sub` is a default provided claim, the RFC states: 
> The "sub" (subject) claim identifies the principal that is the subject of the JWT.

This is one claim that can be mapped to a Redpanda principal.